src/lib/y2network/network_manager/config_writer.rb

```diff
@@ -68,7 +87,21 @@ def write_dns(config, old_config, _issues_list)
       # @param routes [Array<Route>] List of routes to search in
       # @return [Array<Route>] List of routes for the given connection
       def routes_for(conn, routes)
-        routes.select { |r| r.interface&.name == conn.name }
+        return routes.reject(&:interface) if conn.nil?
+
+        explicit = routes.select { |r| r.interface&.name == conn.name }
+
+        return explicit if !conn.static_valid_ip?
```

Comment:
Why not use `unless`?

Reply:
no strong reason, so, just switch to unless